samples/net/sockets/echo_client/overlay-802154.conf

```diff
@@ -4,6 +4,12 @@ CONFIG_BT=n
 CONFIG_NET_TCP=n
 CONFIG_NET_IPV4=n
 
+# Disabling Ethernet
+CONFIG_NET_SLIP_TAP=n
+# Enable UART PIPE 802.15.4 driver
+CONFIG_IEEE802154_UPIPE=y
```

Comment:
I don't think we should enable specific drivers in this overlay in order to keep it generic. So far I was using this overlay with nRF devices, which enables it's 15.4 radio driver automatically based on L2 configuration. The same pattern was followed by `net_app` echo samples, yet they provide separate config for qemu 802.15.4 operation, see https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/net/echo_client/overlay-qemu_802154.conf.

Therefore I think we should either:
* enable `IEEE802154_UPIPE` automatically when qemu and 802154 is used and reuse this overlay,
* provide separate overlay for tests over qemu and enable UPIPE there.

Reply:
oh yes, I mixed up. non-socket based echo app have qemu specific overlay for 15.4. I may create ones for that board then.